dockers/docker-database/supervisord.conf.j2

```diff
@@ -20,7 +20,7 @@ stderr_logfile=syslog
 {% if INSTANCES %}
 {%     for redis_inst, redis_items in INSTANCES.iteritems() %}
 [program: {{ redis_inst }}]
-command=/bin/bash -c "{ [[ -s /var/lib/{{ redis_inst }}/dump.rdb ]] || rm -f /var/lib/{{ redis_inst }}/dump.rdb; } && mkdir -p /var/lib/{{ redis_inst }} && exec /usr/bin/redis-server /etc/redis/redis.conf --port {{ redis_items['port'] }} --unixsocket {{ redis_items['unix_socket_path'] }} --pidfile /var/run/redis/{{ redis_inst }}.pid --dir /var/lib/{{ redis_inst }}"
+command=/bin/bash -c "{ [[ -s /var/lib/{{ redis_inst }}/dump.rdb ]] || rm -f /var/lib/{{ redis_inst }}/dump.rdb; } && mkdir -p /var/lib/{{ redis_inst }} && exec /usr/bin/redis-server /etc/redis/redis.conf --bind {{ LOOPBACK_IP }} {{ redis_items['hostname'] }} --port {{ redis_items['port'] }} --unixsocket {{ redis_items['unix_socket_path'] }} --pidfile /var/run/redis/{{ redis_inst }}.pid --dir /var/lib/{{ redis_inst }}"
```

Comment:
>{{ redis_items['hostname'] }} [](start = 230, length = 29)

What is this for?

Reply:
I have added two IP addresses for bind when we start redis server in the namespaces viz. 127.0.0.1 and 240.1.1.* (address which is on eth0 and connected to docker0 bridge in the host ). 
In the host we still continue to start on the loopback IP 127.0.0.1 as if you notice {{ redis_items['hostname'] }} for host namespace takes the IP address from lo interface.

In the namespace, teh {{ redis_items['hostname'] }} will have the eth0 ip address in 240.* subnet. I explicitly added the  {{ LOOPBACK_IP }} here because I have seen some existing scripts using this IP address hardcoding -- hence need to bind this IP address also while starting redis in namesapce.

Reply:
To clarify, why we need `{{ redis_items['hostname'] }}` ? Why we don't need before?

---
In reply to: [441926664](https://github.com/Azure/sonic-buildimage/pull/4779#discussion_r441926664) [](ancestors = 441926664)

Reply:
This is needed as I am making the this field "hostname" in the database_config.json.j2 a template variable as below 
{
"INSTANCES": {
    "redis":{
        "hostname" : "{{HOST_IP}}",

The HOST_IP derived in docker-database-init.sh will be exported as an environment variable and used to generate the database_config.json file from the  j2 template.

Reply:
I see. Some suggestions:
1. add `--bind` before the parameter
2. add quotations around {{ LOOPBACK_IP }} and {{ redis_items['hostname'] }}, in case anything is missing the cli will catch it easily.

---
In reply to: [443091002](https://github.com/Azure/sonic-buildimage/pull/4779#discussion_r443091002) [](ancestors = 443091002,441926664)

Reply:
In this command string --bind {{ LOOPBACK_IP }} {{ redis_items['hostname'] }} the --bind is already before the parameters. 

For the second comment, I have made sure in docker-database-init.sh we get a valid IP address in the hostname field. If the IP address derivation fails, the loopback ip address 127.0.0.1 is populated by default.

Reply:
What if redis missing the desired elements? Do you want to detect this case  and fail redis-server?

---
In reply to: [443148626](https://github.com/Azure/sonic-buildimage/pull/4779#discussion_r443148626) [](ancestors = 443148626,443091002,441926664)